src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -141,6 +142,30 @@ private Task OnProjectFactoryCompletedAsync()
             }
         }
 
+        private JoinableTask<T> ExecuteWithinLockAsync<T>(Func<Task<T>> task)
+        {
+            return JoinableFactory.RunAsync(async delegate
+            {
+                using (JoinableCollection.Join())
```

Comment:
>using (JoinableCollection.Join()) [](start = 16, length = 33)

Without context this looks very confusing, can you add a comment as to why you need to add the extra using here?